### PR TITLE
fix(communityEdition): fix unprocessable entity error

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:
Fix an issue where garden failed with the error "unprocessable entity" some times when user has more than 24 projects.

Previously core listed all projects and if it didn't find project with given name in the first list of results, it would attempt creating the project. Creating the project fails then with "unprocessable entity" because a project with the given name already exists.

**Special notes for your reviewer**:
Depends on a backend change, thus the draft state

### CODE REVIEW

File: core/src/cloud/api.ts [hunk 1]
```diff
@@ -359,41 +359,38 @@ export class CloudApi {
...
       throw new CloudApiDuplicateProjectsError({
         message: deline`Found an unexpected state with multiple projects using the same name, ${projectName}.
         Please make sure there is only one project with the given name.
         Projects can be deleted through the Garden Cloud UI at ${this.domain}`,
       })
     }
 
-    return projects[0]
+    if (projectList.length === 0) {
+      return undefined
```
Comment (10ko) [inline core/src/cloud/api.ts:385]: Any reason why we return undefined instead of throwing?

Comment (stefreak) [reply core/src/cloud/api.ts:385]: That's the behaviour the calling function [`getOrCreateProject`](https://github.com/garden-io/garden/pull/5931/files#diff-c2e1d8caa9ddda54623fafa053401b23d34f039f71077069cec6a6f563f027d0R419) depends on.

**Review by 10ko — DISMISSED** [review]

File: core/src/cloud/api.ts (hunk 1 above)
Comment (stefreak) [reply core/src/cloud/api.ts:385]: I think it's good to return `undefined` in this case; Using exceptions for control flow would be a bad practice in this case in my opinion.

**Review by 10ko — APPROVED** [review]

**Review by 10ko — DISMISSED** [review]

**Review by twelvemo — APPROVED** [review]
Nice!

**Pull request merged**